src/ring_theory/tensor_product.lean

```diff
@@ -845,3 +845,168 @@ begin
   rw [←E1.range_val, ←E2.range_val, ←algebra.tensor_product.product_map_range],
   exact (algebra.tensor_product.product_map E1.val E2.val).to_linear_map.finite_dimensional_range,
 end
+
+section is_tensor_product
+
+variables {R : Type*} [comm_ring R]
+variables {M₁ M₂ M M' : Type*}
+variables [add_comm_monoid M₁] [add_comm_monoid M₂] [add_comm_monoid M] [add_comm_monoid M']
+variables [module R M₁] [module R M₂] [module R M] [module R M']
+variable (f : M₁ →ₗ[R] M₂ →ₗ[R] M)
+variables {N₁ N₂ N : Type*} [add_comm_monoid N₁] [add_comm_monoid N₂] [add_comm_monoid N]
+variables [module R N₁] [module R N₂] [module R N]
+variable {g : N₁ →ₗ[R] N₂ →ₗ[R] N}
+
+/--
+Given a bilinear map `f : M₁ →ₗ[R] M₂ →ₗ[R] M`, `is_tensor_product f` means that
+`M` is the tensor product of `M₁` and `M₂` via `f`.
+This is defined by requiring the lift `M₁ ⊗[R] M₂ → M` to be bijective.
+-/
+def is_tensor_product : Prop := function.bijective (tensor_product.lift f)
```

Comment:
Would it make sense to follow the design of `direct_sum.decomposition` here and provide an explicit inverse?

Reply:
I don't really understand your comment about canonicity; the inverse is always canonical because it's unique, right?

My thinking behind making the typeclass carry data is that an instance can still always provide it non-computably; but we don't have to throw away computability to use `tensor_product` through this API.

Reply:
I thought the main concern were bad defeqs, and what I was saying that the definition of the inverse is usually no better than "the inverse of the lift".
But still I don't think there are many cases where there is a computable inverse? Maybe `M/IM -> A/I x M` since quotients seem to be computable?
I think the choice I made is in line with the approach we took on localizations: we have a computable `localization` and an `Prop`-valued `is_localization` whose API are mostly incomputable.